### ISSUE TITLE
Deleted Hours of Work

### ISSUE THREAD

**Issue by GirkovArpa** [johanmattssonm/birdfont#120]
I just spent several hours modifying a font, and saving it several times (as a birdfont file), only to have Bird Font delete it as soon as I closed the window.

Where is it at?

I was literally looking the at the file in an explorer window when I closed Bird Font, and the ~45mb file disappeared when Bird Font closed. 🤯

____
I just recovered it with Recuva, but it's only partly recovered and Bird Font opens it as an empty font.

**Comment by johanmattssonm** [comment]
I have never heard of this before. Please check the backups in the settings
tab.

**Comment by GirkovArpa** [comment]
It says there are no backups.  It occured after I got the popup telling me I can't export because of the license, and need to purchase the commercial version.  My first thought was I'm not allowed to even save birdfont files with the free version?

**Comment by johanmattssonm** [comment]
Yes, you are allowed to save your font with all versions of birdfont. Which folder did you save to? Do you see anything listed in the recent fonts tab (Ctrl+R or Command+R)?

**Comment by johanmattssonm** [comment]
Please email me the recovered font. I would like to have a look at it. Address: johan.mattsson.m@gmail.com

**Comment by johanmattssonm** [comment]
I am also very much interested in a way to reproduce this problem if you know of way to make this happen.

**Comment by GirkovArpa** [comment]
Sent.  

What I had done was drag the .birdfont file onto an Frhed.lnk file to open it in the Frhed hex editor while it was also opened in Bird Font.  Frhed gave me an error saying it couldn't open the file, so I closed Frhed and saw that the .birdfont file vanished (I could see it in an explorer window).

There were no backups.

I made a shortcut to Bird Font for convenience on my Desktop right after and immediately recovered the file with Recuva.  Unfortunately this shortcut (along with some other stuff) had already overwritten the top of the .birdfont file in those 60 seconds.  I wasn't quite sure what the gibberish sandwiched between the .lnk file and the tail-end of the .birdfont file was, so I trimmed the top of the .birdfont file off before sending you a link.

**Comment by johanmattssonm** [comment]
Thank you. I am working on it. Did  Frhed  have anything to do with the
disappearing file? I will email you a copy with the things I found in the
file you sent.

Can you make a quick test and see if backups work for a new font file? You
should see a new row in the backup tab every time you save your font.

**Comment by GirkovArpa** [comment]
Thank you for sending back a usable file.  Oddly, it appears to only reflect my very first save yesterday (modifications of 2 characters), but not the modification of about 20 characters I performed today, even though I saved multiple times.

There were 3 images I believe, but I have those saved elsewhere.

Backups are working normally now.

To answer your question about Frhed:  

I considered the very remote possibility that Frhed has a "Delete File" action, and I accidentally clicked it or its hotkey when trying to close it.    I didn't really expect any program to the power to straight up delete the currently open file without even sending it to the Recycle Bin, but ...

![](https://i.imgur.com/LzYc3Me.png)

I apologize for accusing your innocent program of deleting my file!
